lib/player/player.js

```diff
@@ -736,6 +741,36 @@ shaka.player.Player.prototype.setPlaybackRate = function(rate) {
 
 
 /**
+ * @param {Object} restrictions An Object containing the restrictions
+ *     on video tracks that you wish to set. For example, passing
+ *     {'maxBandwidth': 700000, 'minBandwidth': 200000} will restrict
+ *     playback to video tracks with bandwidth between 700000 and 200000.
+ * @export
+ */
+shaka.player.Player.prototype.setRestrictions = function(restrictions) {
+  for (var key in restrictions) {
```

Comment:
Not by the manifest.  There aren't any restrictions in the manifest.

The license response may have restrictions in the form of a app-level header, but even that is an application-specific convention arranged between the client app and its own license server.  So either way, it's application-specific or service-provider-specific logic that sets restrictions, and up to the application to set them intelligently.  The library simply gives the application a chance to set restrictions based on the license response, since this is a pre-existing paradigm that we would like to enable.